### PR TITLE
feat: reorganize CLI under new subcommands

### PR DESCRIPTION
**Pull request by chadoh**
### What

Adds `contract` and `labs` subcommands, nesting relevant commands within, as [proposed](https://docs.google.com/document/d/16lJhyMU6cqRzNtMclr9yXd978MJgbZ0El9UhxbTz_Ts/edit#).

```txt
$ soroban --help
soroban 0.3.1
https://soroban.stellar.org

USAGE:
    soroban <SUBCOMMAND>

OPTIONS:
    -h, --help    Print help information

SUBCOMMANDS:
    contract      Tools for smart contract developers
    serve         Run a local webserver for web app development and testing
    lab           Experiment with early features and expert tools
    version       Print version information
    completion    Print shell completion code for the specified shell


$ soroban contract --help
Tools for smart contract developers

USAGE:
    soroban contract <SUBCOMMAND>

OPTIONS:
    -h, --help    Print help information

SUBCOMMANDS:
    bindings    Generate code client bindings for a contract
    deploy      Deploy a contract
    inspect     Inspect a WASM file listing contract functions, meta, etc
    install     Install a WASM file to the ledger without creating a contract instance
    invoke      Invoke a contract function
    optimize    Optimize a WASM file
    read        Print the current value of a contract-data ledger entry


$ soroban lab --help
Experiment with early features and expert tools

USAGE:
    soroban lab <SUBCOMMAND>

OPTIONS:
    -h, --help    Print help information

SUBCOMMANDS:
    token    Wrap, create, and manage token contracts
    xdr      Decode xdr
```

### Why

Improve developer onboarding by making the CLI more educational.

### Known limitations

Does not add new proposed features, only reorganizes existing commands.

### CODE REVIEW

**Review by paulbellamy — APPROVED** [review]

File: cmd/soroban-cli/src/contract/invoke.rs [hunk 1]
```diff
@@ -41,9 +41,6 @@ pub struct Cmd {
     /// Argument to pass to the function
     #[clap(long = "arg", value_name = "arg", multiple = true)]
     args: Vec<String>,
-    /// Argument to pass to the function (base64-encoded xdr)
-    #[clap(long = "arg-xdr", value_name = "arg-xdr", multiple = true)]
-    args_xdr: Vec<String>,
```
Comment (paulbellamy) [inline cmd/soroban-cli/src/contract/invoke.rs:45]: I worry about this, as there are some args (for example `Identifier`s), which are super difficult to pass via the `--arg` flag.

Comment (paulbellamy) [reply cmd/soroban-cli/src/contract/invoke.rs:45]: ^ is this true @tyvdh ?

Comment (tyvdh) [reply cmd/soroban-cli/src/contract/invoke.rs:45]: I'll defer to @willemneal but if his PR around arg improvements goes live we may not need it. I definitely want to ensure all possible args are able to be supported but I'm under the impression his PR will do that

https://github.com/stellar/soroban-tools/pull/307
https://github.com/stellar/soroban-tools/pull/284

Comment (paulbellamy) [reply cmd/soroban-cli/src/contract/invoke.rs:45]: Ok then. #307 looks good as well, so let's do it.

Comment (chadoh) [reply cmd/soroban-cli/src/contract/invoke.rs:45]: Pro tip, @tyvdh: list PRs in bulleted lists and GitHub expands them to show status and title. You could edit your comment to just add `- ` at the beginning of each of your PR lines.

- #307
- #284

**Pull request merged**